templates/src/main/java/org/hawkular/commons/templates/internal/TemplateServiceImpl.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2014-2015 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.commons.templates.internal;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.util.LinkedHashSet;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+
+import javax.annotation.security.PermitAll;
+import javax.ejb.Singleton;
+import javax.inject.Inject;
+
+import org.hawkular.commons.templates.TemplateService;
+
+import freemarker.cache.FileTemplateLoader;
+import freemarker.cache.MultiTemplateLoader;
+import freemarker.cache.TemplateLoader;
+import freemarker.template.Configuration;
+import freemarker.template.Template;
+import freemarker.template.TemplateException;
+
+/**
+ * @author Juraci Paixão Kröhling
+ */
+@PermitAll
+@Singleton
+public class TemplateServiceImpl implements TemplateService {
+    @Inject @OverrideDirectory
+    String overrideDirectory;
+
+    @Inject @ConfigurationDirectory
+    String configurationDirectory;
+
+    private Configuration configuration = new Configuration(Configuration.VERSION_2_3_23);
+    private boolean isConfigured = false;
+
+    public TemplateServiceImpl() throws IOException {
```

Comment:
Why not do this in a @PostConstruct method and remove the isConfigured checks?

If it's not configured it shouldn't be used anyway.